tsconfig.json

```diff
@@ -0,0 +1,47 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "server/*": [
+        "server/*"
+      ],
+      "client/*": [
+        "client/*"
+      ]
+    },
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "jsx": "preserve",
+    "lib": [
+      "dom",
+      "es2017"
+    ],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noEmit": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "preserveConstEnums": true,
+    "removeComments": false,
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "strict": true,
+    "target": "es6",
```

Comment:
We should review whether we should decide on `es6` or `es5` for extended backwards compatibility. Generally, I'd target `es5`.

Reply:
Do we still need to specify the target if we are using Babel?

Reply:
From what I've read, if we decide to go for module: `esnext` then we could also set the target to `esnext`